build.py

```diff
@@ -299,7 +299,7 @@ def _validate_cmake_args(args: argparse.Namespace):
 
 def _validate_args(args: argparse.Namespace):
     # default to all 3 stages
-    if not args.update and not args.build and not args.test:
+    if not args.update and not args.build and not args.test and not args.clean:
```

Comment:
I think args.clean should override all other args. For example, if user provided --clean --build, we should probably only clean (or raise an error or warning).
Or alternatively, clean first and then build again? We probably should document the expected behavior.

Reply:
Agreed. Will fix

Reply:
Is clean supposed remove the build directory for the specified or default config? doesn't appear to be doing so

Reply:
Apparently cmake clean only removes the cmake generated files. I wanted to remove all of the previous build artifacts

Reply:
Updated to remove the build folder as well